GridList/GridList.jsx

```diff
@@ -0,0 +1,178 @@
+import { h } from "preact";
+import MaterialComponent from "../MaterialComponent";
+
+/**
+ * @prop cellHeight	{number} - Number of px for one cell height. You can set 'auto' if you want to let the children determine the height.
+ * @prop cols {number} - Number of columns.
+ * @prop padding {1|4} - Number of px for the padding/spacing between items.
+ * @prop style {object} - Override the inline styles of the root element
+ */
+class GridList extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "grid-list";
+    this._propsDict = {
+      cols: "cols",
+      padding: "padding",
+      cellHeight: "cellHeight",
+      secondaryActionIconAlign: "secondaryActionIconAlign",
+      aspectRatio: "aspectRatio",
+      twoLineCaptions: "twoLineCaptions"
+    };
+  }
+  createClassName(props) {
+    const baseClass = "mdc-grid-list";
+    const classes = [];
+
+    if (
+      props[this._propsDict.padding] &&
+      props[this._propsDict.padding] === 1
+    ) {
+      classes.push("mdc-grid-list--tile-gutter-1");
+    }
+
+    if (props[this._propsDict.secondaryActionIconAlign]) {
+      classes.push(
+        "mdc-grid-list--with-icon-align-" +
+          props[this._propsDict.secondaryActionIconAlign]
+      );
+    }
+
+    if (props[this._propsDict.aspectRatio]) {
+      classes.push(
+        "mdc-grid-list--tile-aspect-" + props[this._propsDict.aspectRatio]
+      );
+    }
+
+    if (props[this._propsDict.twoLineCaptions]) {
+      classes.push("mdc-grid-list--twoline-caption");
```

Comment:
Please check https://github.com/prateekbh/preact-material-components/pull/247#issuecomment-327836903 for more information on how to implement this correctly.

Reply:
That is indeed more elegant, but I think I want to clarify one thing: it appears that `_mdcProps` is a way to allow for easy boolean-based classname setting?  In that case, it would a perfect fit for

```
    if (props[this._propsDict.twoLineCaptions]) {
      classes.push("mdc-grid-list--twoline-caption");
    }
```

Since that is boolean, it could be replaced with `this._mdcProps = ["twoline-caption"];` and then could be activated by defining an instance as `<GridList twoline-caption=true>`.

However, for tile-gutter-*, tile-aspect-*, and with-tile-icon-*, since the `*` is a placeholder for a value, I'm assuming we still need something like the `createClassName` method correct?

Reply:
@filoozom did I understand how `_mdcProps` works?